presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -247,10 +277,150 @@ private String indentString(int indent)
         return Strings.repeat("    ", indent);
     }
 
-    private static class AnySymbolReference
-            extends SymbolReference
+    public static SymbolStem symbolStem(String stem)
     {
-        AnySymbolReference()
+        return new SymbolStem(stem);
+    }
+
+    /*
+     * All comments below about Symbols apply to SymbolReferences as well.
+     *
+     * This is the result of a compromise to satisfy as many of the following
+     * constraints as possible:
+     *
+     * 1) Many objects we want to compare in plans already define equals
+     * methods. We should use these to the greatest degree possible in the
+     * various Matchers, because there's no test-specific code that can get
+     * out of date.
+     * 2) Many of said objects contain either Symbols.
+     * 3) Symbols in the final Plan may be decorated by the SymbolAllocator to
+     * make them unique as needed.
+     * 4) There's no way to know a-priori what the decorated name of a Symbol
+     * will be in the plan. Even if we could, future changes to the planner
+     * might change the name of the Symbols in the plan, rendering the expected
+     * values in tests that depended on them out of date.
+     *
+     * The solution to writing expected plans that match actual plans and are
+     * as self-maintaining as possible is to extend Symbol so that the
+     * following returns true:
+     *   expectedSymbol.equals(actualDecoratedSymbol)
+     *
+     * This ensures that we can construct expected objects containing Symbols
+     * and use their built-in equals() methods to compare them to the actual
+     * objects in the plan.
+     *
+     * To actually implement this, we need to handle comparing the equality
```

Comment:
This currently implicitly depends on the behavior of the SymbolAllocator. I'm happy to make the dependency explicit by creating an `@VisibleForTesting` public static method in SymbolAllocator that demangles symbols, then adding unit tests to TestSymbolAllocator to make this dependency explicit.